### PR TITLE
[DPE-5680] Fix handling of azure storage relation databag and fix CI

### PR DESCRIPTION
**Pull request by theoctober19th**

### CODE REVIEW

**Review by deusebio — COMMENTED** [review]
Looks very good! Would it be easy and possible to add a unit test to check the None behaviour?

File: pyproject.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ authors = []
 
 [tool.poetry.dependencies]
 python = "^3.10"
-ops = "^2.11.0"
+ops = "^2.17.0"
```
Comment (theoctober19th) [inline pyproject.toml:13]: reason: `ops` did not previously have support for `type: secret` in their testing framework.

**Review by deusebio — APPROVED** [review]
Very nice! Great job!

**Review by welpaolo — APPROVED** [review]
